#!/usr/bin/env python

PACKAGE = "allegro_opposition_primitives"
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
 #       Name       Type      Level Description     Default Min   Max
grasp_enum = gen.enum([	gen.const("grasp_none", str_t, "grasp_none", "Home position"),
						gen.const("thumb_surface_opp_index_side", str_t, "tsOs1", "Thumb surface - Index side"),
						gen.const("thumb_tip_opp_index_middle_tip", str_t, "ttOft12", "Thumb tip - Index Middle tip"),
						gen.const("power_with_tip_stabilize", str_t, "power_with_tip_stabilize", "Grasp 1")], "Grasp choice")
gen.add("grasp", str_t, 0, "Grasp choice", "grasp_none", edit_method=grasp_enum)

gen.add("close", bool_t, 0, "Close the grasp", False)

primitive_enum = gen.enum([	gen.const("p0", int_t, 0, "primtive 0"),
							gen.const("p1", int_t, 1, "primtive 1"),
							gen.const("p2", int_t, 2, "primtive 2"),
							gen.const("p3", int_t, 3, "primtive 3"),
							gen.const("p4", int_t, 4, "primtive 4"),
							gen.const("p5", int_t, 5, "primtive 5"),
							gen.const("p6", int_t, 6, "primtive 6")], "The active primitive")
gen.add("primitive", int_t, 0, "The active primitive", 0, 0, 6, edit_method=primitive_enum)
gen.add("mix_factor", double_t, 0, "Importance of the primitive", 1, 0, 1)
gen.add("close_level", double_t, 0, "Governs speed of closing", 0.07, 0.001, 0.2)
gen.add("squeeze_level", double_t, 0, "Governs magnitude of directional internal forces", 0.07, 0.001, 0.6)


exit(gen.generate(PACKAGE, "allegro_opposition_primitives", "my_dyn_params"))
